spec/factories/whitehall_export/edition_factory.rb

```diff
@@ -52,5 +52,28 @@
         ]
       end
     end
+
+    trait :published do
+      transient do
+        previous_state { "submitted" }
+      end
+
+      created_at { 3.days.ago.rfc3339 }
+      scheduled_publication { Time.current.tomorrow.rfc3339 }
+      state { "published" }
+      revision_history do
+        [
+          build(:revision_history_event, created_at: created_at),
+          build(:revision_history_event,
+                event: "update",
+                created_at: 2.days.ago.rfc3339,
+                state: previous_state),
+          build(:revision_history_event,
+                event: "update",
+                state: "published",
+                created_at: Time.current.tomorrow.rfc3339),
+        ]
+      end
```

Comment:
I see there's other scheduled bits in here. I wonder if it'd work the same for us were we to do `build(:whitehall_export_edition, state: "published")` and avoid needing the trait?

Reply:
I tried that, but it doesn't work because you also need a revision history entry for the "published" state, so I thought it was nicer to create a trait instead.

Reply:
No problem 👍